policy-controller/k8s/status/src/http_route.rs

```diff
@@ -5,20 +5,21 @@ use linkerd_policy_controller_k8s_api::{
     policy::{self, Server},
 };
 
-/// Represents an HTTPRoute's parent reference from its spec.
+/// Represents an HTTPRoute's reference from its spec.
 ///
-/// This is separate from the policy controller index's `InboundParentRef`
-/// because it does not validate that the parent reference is not in another
-/// namespace. This is something that should be relaxed in the future in the
-/// policy controller's index and we could then consider consolidating these
-/// types into a single shared lib.
+/// Each HTTPRoute may have a number of parent or backend references.
+/// Reference serves as a wrapper around one these reference's identifier and
+/// concrete type.
 #[derive(Clone, Eq, PartialEq)]
-pub enum ParentReference {
-    Server(ResourceId),
+pub struct Reference {
```

Comment:
That looks like it'd fit our use cases, I'm all for it. Just to play devil's advocate a little bit, rolling out our own type has the advantage of keeping _most_ of the status controller logic in this crate. If we want to define any additional associated functions on the type we'll have to navigate between crate definitions and that may induce a bit more cognitive burden. I'm not sure if that's enough of an argument to advise against it though.

I'm easy either way.

Reply:
Yea `NamespacedTargetRef` would work well here. We'll have to add some methods to it in that crate, but using it for parent and backend references feels like the right move rather than duplicating a lot of that logic here.

@adleong to answer your review question here, using one type for both of these references does feel like the right thing to do here. Just to be clear, we're not combining parent and backend references in a single `Vec`; once we start supporting backend references they'll be in a separate `Vec`. But yes they'll ultimately use the same type.

Any separate logic that needs to happen for creating conditions will be separate by the fact that we're working with either parent or backend references.